### PR TITLE
Spark chart refactor

### PR DESCRIPTION
**Pull request by meyskens**
This is a total refactor of the Spark chart.
It breaks completely with the old chart and with the official chart.
The chart is composed to follow common practiches in Helm Charts.

Signed-off-by: Maartje Eyskens <maartje@eyskens.me>

### CODE REVIEW

File: spark/templates/spark-webui-proxy-service.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+apiVersion: v1
+metadata:
+  name: {{ template "webui-proxy-fullname" . }}
+spec:
+  ports:
+    - port: {{ .Values.webUIProxy.service.port }}
+      targetPort: http
+      {{- if .Values.webUIProxy.service.nodePort }}
+      nodePort: {{ .Values.webUIProxy.service.nodePort }}
+      {{- end }}
```
Comment (rporres) [inline spark/templates/spark-webui-proxy-service.yaml]: Looks kinda weird that only this service supports `nodePort` and the rest do not.

File: spark/README.md [hunk 2]
```diff
@@ -1,133 +0,0 @@
-# Apache Spark Helm Chart
```
Comment (rporres) [inline spark/README.md]: I understand we don't want to maintain config, but I would keep this README with the deleted configuration

File: spark/templates/_helpers.tpl [hunk 3]
```diff
@@ -12,30 +12,25 @@ We truncate at 63 chars because some Kubernetes name fields are limited to this
...
 {{- define "webui-fullname" -}}
 {{- $name := default .Chart.Name .Values.nameOverride -}}
-{{- printf "%s-%s-%s" .Release.Name $name .Values.WebUi.Name | trunc 63 | trimSuffix "-" -}}
+{{- printf "%s-%s-webui" .Release.Name $name | trunc 63 | trimSuffix "-" -}}
 {{- end -}}
 
 {{- define "livy-fullname" -}}
 {{- $name := default .Chart.Name .Values.nameOverride -}}
-{{- printf "%s-%s-%s" .Release.Name $name .Values.Livy.Name | trunc 63 | trimSuffix "-" -}}
+{{- printf "%s-%s-livly" .Release.Name $name | trunc 63 | trimSuffix "-" -}}
```
Comment (rporres) [inline spark/templates/_helpers.tpl]:
```suggestion
{{- printf "%s-%s-livy" .Release.Name $name | trunc 63 | trimSuffix "-" -}}
```

File: spark/templates/configmap.yaml [hunk 4]
```diff
@@ -1,4 +1,4 @@
-{{- if .Values.Livy.Enabled }}
```
Comment (rporres) [inline spark/templates/configmap.yaml]: There are things to change inside this config map that refer to the old values names

File: spark/values.yaml [hunk 5]
```diff
@@ -3,91 +3,81 @@
...
   # Additional volume mounts for the Master container
   additionalVolumeMounts: []
-  # Custom Spark Config. See template to find which values are already set
-  customSparkConfig: {}
+  extraEnv: []
   # Set Master JVM memory. Default 1g
-  # DaemonMemory: 1g
-  # NodeSelector: srcd.host/type: service
+  # daemonMemory: 1g
+  # dodeSelector: srcd.host/type: service
```
Comment (driosalido) [inline spark/values.yaml]:
```suggestion
  # nodeSelector: srcd.host/type: service
```

File: spark/templates/spark-webui-proxy-service.yaml (hunk 1 above)
Comment (meyskens) [reply spark/templates/spark-webui-proxy-service.yaml]: You're right! I overlooked it for the others

File: spark/templates/configmap.yaml (hunk 4 above)
Comment (meyskens) [reply spark/templates/configmap.yaml]: added a `checksum/configmap`

File: spark/templates/spark-webui-proxy-service.yaml (hunk 1 above)
Comment (meyskens) [reply spark/templates/spark-webui-proxy-service.yaml]: Added them

File: spark/README.md (hunk 2 above)
Comment (meyskens) [reply spark/README.md]: Do you mean the readme but without the configuration part?

Comment (rporres) [reply spark/README.md]: yes!

**Pull request merged**